lib/util/defineRuleString.js

```diff
@@ -0,0 +1,24 @@
+const {definePatternRule} = require('math-rules');
+const {parse} = require('math-parser');
+const {traverse} = require('math-traverse');
+
+const defineRuleString = (matchPattern, rewritePattern, constraints) => {
+  const matchAST = parse(matchPattern);
+  const rewriteAST = parse(rewritePattern);
+
+  traverse(matchAST, {
```

Comment:
Why do you delete `loc` when defining the rule? It seems like it'd make more sense to do this just once at the beginning of a step before trying all the rule matches.

Or maybe I don't understand where `loc` gets set